code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityStorageManager.java

```diff
@@ -0,0 +1,134 @@
+/*
+  Copyright 2021 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.edge.identity;
+
+import static com.adobe.marketing.mobile.edge.identity.IdentityConstants.LOG_TAG;
+
+import androidx.annotation.VisibleForTesting;
+import com.adobe.marketing.mobile.services.DataStoring;
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.services.NamedCollection;
+import com.adobe.marketing.mobile.services.ServiceProvider;
+import com.adobe.marketing.mobile.util.JSONUtils;
+import com.adobe.marketing.mobile.util.StringUtils;
+import java.util.Map;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Manages persistence for this Identity extension
+ */
+class IdentityStorageManager {
+
+	private static final String LOG_SOURCE = "IdentityStorageService";
+	private final NamedCollection edgeIdentityStore;
+	private final NamedCollection directIdentityStore;
+
+	IdentityStorageManager() {
+		this(ServiceProvider.getInstance().getDataStoreService());
+	}
+
+	@VisibleForTesting
+	IdentityStorageManager(final DataStoring dataStoreService) {
+		this.edgeIdentityStore = dataStoreService.getNamedCollection(IdentityConstants.DataStoreKey.DATASTORE_NAME);
+		this.directIdentityStore =
+			dataStoreService.getNamedCollection(IdentityConstants.DataStoreKey.IDENTITY_DIRECT_DATASTORE_NAME);
+	}
+
+	/**
+	 * Loads identity properties from local storage, returns null if not found.
+	 *
+	 * @return properties stored in local storage if present, otherwise null.
```

Comment:
it also returns null in case of a serialization error, can you update the description with more details